### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by potocpav**
Changed some minor stuff, mainly switched the busy-wait framerate limitation for vertical sync. The animations are smoother and the CPU happier :)

### CODE REVIEW

File: shaders/fragment.glsl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#version 330
+#version 130
```
Comment (camjackson) [inline shaders/fragment.glsl:1]: Yeah, as @Dr-Emann mentioned, I don't think this version is supported on newer Macs, which is why I bumped it up.

**Comment by camjackson** [comment]
Hmm, those comments are a bit context-less here. I should have added them to specific lines instead of each commit.

I like most of these changes though! The shader version I'd like to leave at 330, and the timing change I'm not sure about, as I think the simulation speed should be fixed, and independent of rendering. Open to discussion though :)

**Comment by camjackson** [comment]
Update: 
- I'd like to leave the shader version at 330
- I liked your change to the seed data, but I ended up merging @Dr-Emann's implementation of that idea instead
- I've cherry-picked your fix of the tests (thanks!)
- I'm still not sure about the vsync thing, but I may have misunderstood the implications of your change
- I like your change to move the grid constructor into the grid impl, but unfortunately it doesn't merge now, even as a cherry-pick :( If you fix the conflict and open a separate pull request for just that change, I'll happily accept it. Or I can do it if you like, but you'll lose the commit credit :P

**Comment by potocpav** [comment]
OK, leave the version number. Guess I have to buy a new laptop :D 

I've noticed now that I misunderstood the behaviour of the main loop and wrongly changed it to limit the update rate.  On average, half the update time is added to the render time; that is the reason the animation seemed so sluggish to me (on my crappy laptop) and I decided to fix it :) The correct solution would be to limit rendering to vsync and update the grid in a separate thread.

That's probably it for this PR. I will shortly submit another one with the grid constructor.

**Comment by camjackson** [comment]
The lowest shader version that works for me is 140. Does that work for you?

**Comment by Dr-Emann** [comment]
140 works for me.

**Comment by camjackson** [comment]
@potocpav I've reduced the shader versions down to 140, which is the lowest that my laptop supports. Let me know if that works on your machine.

**Comment by potocpav** [comment]
@camjackson  No, it does not work, though it's likely a compatibility improvement.

``` rust
CompilationError("0:1(10): error: GLSL 1.40 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.00 ES, and 3.00 ES\n")
```

**Pull request closed without merging**